test/snippet/core/algorithm/configuration_combine.cpp

```diff
@@ -0,0 +1,40 @@
+#include <seqan3/core/algorithm/configuration.hpp>
+#include <seqan3/core/algorithm/pipeable_config_element.hpp>
+
+// Initial setup used in the actual example:
+enum struct my_id : int
+{
+    bar_id,
+    foo_id
+};
+
+struct bar : public seqan3::pipeable_config_element<bar, float>
+{
+    static constexpr my_id id{my_id::bar_id};
+};
+
+template <typename t>
+struct foo : public seqan3::pipeable_config_element<foo<t>, t>
+{
+    static constexpr my_id id{my_id::foo_id};
+};
+
+template <typename t>
+foo(t) -> foo<t>;
+
+namespace seqan3::detail
+{
+template <>
+inline constexpr std::array<std::array<int, 2>, 2> compatibility_table<my_id>
+{
+    {
+        {0, 1},
+        {1, 0}
+    }
+};
+} // namespace seqan3::detail
```

Comment:
@rrahn Should I do this snippet another way?

Reply:
This needs to be discussed with @rrahn. The user documentation should not have references to the detail namespace. In any case users should never open seqan3:: or seqan3::detail. It seems that the underlying design is problematic.

Reply:
Bump @rrahn any suggestions?

Reply:
this is not a user documentation but this snippet is for developers only. Or at least it should not. It is also not a customisation point since it is strongly coupled to the implementation of the algorithm. So I think for the purpose it is fine.

Reply:
It appears in user documentation.

Reply:
Then the place where it is included in user space must be fixed. This is only for seqan developers  who want to write a new configuration for a new algorithm

Reply:
Ah ok. I had a second look, where this occurs. No the snippet should be adapted. Instead use as an example the alignment or search configurations to demonstrate that they are combinable.